source/Engine.cpp

```diff
@@ -888,8 +888,12 @@ void Engine::Step(bool isActive)
 	{
 		double width = max(target->Width(), target->Height());
 		Point pos = target->Position() - center;
-		statuses.emplace_back(pos, flagship->OutfitScanFraction(), flagship->CargoScanFraction(),
-			0., 10. + max(20., width * .5), 4, 1.f, Angle(pos).Degrees() + 180.);
+		const bool outfitBeyondRange = pos.LengthSquared() > (flagship->Attributes().Get("outfit scan power") * 10000);
+		statuses.emplace_back(pos, flagship->OutfitScanFraction(), 0.,
+			0., 10. + max(20., width * .5), 4 + outfitBeyondRange, 1.f, Angle(pos).Degrees() + 180.);
```

Comment:
Instead of adding a magic 4 + boolean, make it a variable, something like
```cpp
int outfitScanColorIndex = (pos.LengthSquared() > /*snip*/) ? 5 : 4;
```
and use that directly.

Reply:
Maybe add a comment too

Reply:
Given #10650 I'd rather just continue with the existing approach rather than assigning meaningful names here just to undo that with that PR (if that PR is merged first, the example you've given won't even work here anymore).

Reply:
Yeah, the entire base setup is horrendous.

Reply:
> Given #10650 I'd rather just continue with the existing approach rather than assigning meaningful names here just to undo that with that PR (if that PR is merged first, the example you've given won't even work here anymore).

Wouldn't my suggested approach be even more compatible with your PR, though? Then instead of 5 and 4, we can put actual names, and still onky need to cast one value.